internal/humanitec/templates.go

```diff
@@ -65,22 +69,31 @@ func (ctx *templatesContext) SubstituteAll(src map[string]interface{}) map[strin
 
 // Substitute replaces all matching '${...}' templates in a source string
 func (ctx *templatesContext) Substitute(src string) string {
-	return os.Expand(src, ctx.mapVar)
+	return placeholderRegEx.ReplaceAllStringFunc(src, func(str string) string {
+		// WORKAROUND: ReplaceAllStringFunc(..) does not provide match details
+		//             https://github.com/golang/go/issues/5690
+		var matches = placeholderRegEx.FindStringSubmatch(str)
+
+		// SANITY CHECK
+		if len(matches) != 3 {
+			log.Printf("Error: could not find a proper match in previously captured string fragment")
+			return src
+		}
+
+		// EDGE CASE: Captures "$$" sequences and empty templates "${}"
```

Comment:
Score placeholders follow the Humanitec syntax: https://developer.humanitec.com/platform-orchestrator/reference/placeholders/

This does not support `$$` for escaping the placeholder, rather `$\{`.

There is a further problem here that if I escape `$${shared.my-var}` it will turn into `${shared.my-var}` and so be interpreted as a valid placeholder by Humanitec which it should not. It should still be escaped for Humanitec to deal with in the delta.

Reply:
`$$` -> `$` is supported right now in all our CLI tools. If we remove this functionality, it may break existing setups, where someone is already using `$$`. We are solving a concrete issue in this PR. Lets stick to it and do not add new breaking changes. We can address all other improvements in separate PRs.

Reply:
But the second point still stands. You *should not* unescape yet. What you need to do is convert it into `$\{` so that Humanitec does not read it.

Reply:
This is not the case. Several placeholders in Humanitec does not have their counterparts in Score resources, and this construct is used exactly to pass through the placeholder as is, so Humanitec would handle it properly. For example, `$${pod.status.hostIP}`.

Note, that similar mechanism is used in `score-compose` to reference environment variables that should not be substituted by the CLI tool, but they are known to `compose` CLI tool.

We can change this behaviour, but again, this would be a breaking change and it does not relate to the issue we are addressing in this PR. A concrete issue is that we want to ignore `$abc` and only react to `${abc}` (with brackets).

Reply:
Let's open a second issue to straighten up escaping more generally.